daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/SchemaComponentIncludesAndImportsMixin.scala

```diff
@@ -35,4 +35,6 @@ trait SchemaComponentIncludesAndImportsMixin extends CommonContextMixin {
     xmlSchemaDocument.uriString
   }.toOption.getOrElse(orElseURL)
```

Comment:
I don't think this trait goes away since a lot of SchemaComponents extend it, and it seems like a straightforward way to set diagnosticFilepath. I do think uriString can go away as we don't seem to be using it anymore

Reply:
Just a note, schemaLocation.uriString is needed still. But the schemaComponents.uriString is not